File: src/BaekJoon/Java/baekjoon_1924.java
package BaekJoon.Java;
// https://www.acmicpc.net/problem/1924

import java.util.Scanner;

public class baekjoon_1924 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int m = sc.nextInt();
		int d = sc.nextInt();
		String[] date = {"SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"};
		int[] day = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

		for(int i = 0; i < m - 1; i++)
			d += day[i];
		d %= 7;

		System.out.println(date[d]);
		sc.close();
	}
}

File: src/BaekJoon/Rust/baekjoon_1924.rs
use std::io::stdin;

pub fn baekjoon_1924() {
    const DAY_OF_WEEK: [&str; 7] = ["SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"];
    const DAY_OF_MONTH: [i32; 12] = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    let month_day: Vec<i32> = stdin().lines().next().unwrap().unwrap()
        .split_whitespace().map(|s| s.parse::<i32>().unwrap()).collect::<Vec<i32>>();
    let (month, mut day) = (month_day[0], month_day[1]);
    
    for index in 0..(month - 1) {
        day += DAY_OF_MONTH[index as usize];
    }
    
    day %= 7;
    
    println!("{}", DAY_OF_WEEK[day as usize]);
}